### ISSUE TITLE
🐛[Android] Camera app crashes on long click on record button 

### ISSUE THREAD

**Issue by UlianaVarlamova** [cuvent/react-native-vision-camera#2652]
### What's happening?

Camera app crashes on a long click on the record button - the click should last about 0,5 second - not short enough for a photo and not long enough for a video. 

**Steps to reproduce:**

Scenario 1:
1. Build Camera app according to the Readme
2. Open app -> Make a long click ( ~0,5 sec ) on the "Photo/Record" button 

Scenario  2:
1. Build Camera app according to the Readme
2. Record a video with a long press.
After some time, say 20-30 seconds of recording, remove your hand from the recording and press it again once or twice.

Actual result for Scenarios 1 and 2:
Camera app is crashed. 
<img width="384" alt="Screenshot 2024-03-14 at 15 01 04" src="https://github.com/mrousavy/react-native-vision-camera/assets/96615514/ecc7bffa-034b-41a6-aed1-3a437839c5b6">

**All logs:** 
[CameraApp.txt](https://github.com/mrousavy/react-native-vision-camera/files/14602605/CameraApp.txt)
**Metro logs:**
[CameraApp_Metro_logs.txt](https://github.com/mrousavy/react-native-vision-camera/files/14602663/CameraApp_Metro_logs.txt)
**Record:**
https://github.com/mrousavy/react-native-vision-camera/assets/96615514/1f23d989-791b-4893-9cd0-b32c89839e79

### Reproduceable Code

```tsx
n/a
```

### Relevant log output

```shell
--------- beginning of crash
03-14 14:44:19.961  4766  5012 E AndroidRuntime: FATAL EXCEPTION: mrousavy/VisionCamera.main
03-14 14:44:19.961  4766  5012 E AndroidRuntime: Process: com.mrousavy.camera.example, PID: 4766
03-14 14:44:19.961  4766  5012 E AndroidRuntime: java.lang.IllegalStateException
03-14 14:44:19.961  4766  5012 E AndroidRuntime: 	at android.media.MediaRecorder.stop(Native Method)
03-14 14:44:19.961  4766  5012 E AndroidRuntime: 	at com.mrousavy.camera.core.RecordingSession.stop(RecordingSession.kt:109)
03-14 14:44:19.961  4766  5012 E AndroidRuntime: 	at com.mrousavy.camera.core.RecordingSession._init_$lambda$0(RecordingSession.kt:78)
03-14 14:44:19.961  4766  5012 E AndroidRuntime: 	at com.mrousavy.camera.core.RecordingSession.$r8$lambda$enBPf4dA0r-MTevnd6ChW75ZCDg(Unknown Source:0)
03-14 14:44:19.961  4766  5012 E AndroidRuntime: 	at com.mrousavy.camera.core.RecordingSession$$ExternalSyntheticLambda0.onError(Unknown Source:2)
03-14 14:44:19.961  4766  5012 E AndroidRuntime: 	at android.media.MediaRecorder$EventHandler.handleMessage(MediaRecorder.java:1615)
03-14 14:44:19.961  4766  5012 E AndroidRuntime: 	at android.os.Handler.dispatchMessage(Handler.java:106)
03-14 14:44:19.961  4766  5012 E AndroidRuntime: 	at android.os.Looper.loopOnce(Looper.java:205)
03-14 14:44:19.961  4766  5012 E AndroidRuntime: 	at android.os.Looper.loop(Looper.java:294)
03-14 14:44:19.961  4766  5012 E AndroidRuntime: 	at android.os.HandlerThread.run(HandlerThread.java:67)
03-14 14:44:29.873  6763  6876 E AndroidRuntime: FATAL EXCEPTION: mrousavy/VisionCamera.main
03-14 14:44:29.873  6763  6876 E AndroidRuntime: Process: com.mrousavy.camera.example, PID: 6763
03-14 14:44:29.873  6763  6876 E AndroidRuntime: java.lang.IllegalStateException
03-14 14:44:29.873  6763  6876 E AndroidRuntime: 	at android.media.MediaRecorder.stop(Native Method)
03-14 14:44:29.873  6763  6876 E AndroidRuntime: 	at com.mrousavy.camera.core.RecordingSession.stop(RecordingSession.kt:109)
03-14 14:44:29.873  6763  6876 E AndroidRuntime: 	at com.mrousavy.camera.core.RecordingSession._init_$lambda$0(RecordingSession.kt:78)
03-14 14:44:29.873  6763  6876 E AndroidRuntime: 	at com.mrousavy.camera.core.RecordingSession.$r8$lambda$enBPf4dA0r-MTevnd6ChW75ZCDg(Unknown Source:0)
03-14 14:44:29.873  6763  6876 E AndroidRuntime: 	at com.mrousavy.camera.core.RecordingSession$$ExternalSyntheticLambda0.onError(Unknown Source:2)
03-14 14:44:29.873  6763  6876 E AndroidRuntime: 	at android.media.MediaRecorder$EventHandler.handleMessage(MediaRecorder.java:1615)
03-14 14:44:29.873  6763  6876 E AndroidRuntime: 	at android.os.Handler.dispatchMessage(Handler.java:106)
03-14 14:44:29.873  6763  6876 E AndroidRuntime: 	at android.os.Looper.loopOnce(Looper.java:205)
03-14 14:44:29.873  6763  6876 E AndroidRuntime: 	at android.os.Looper.loop(Looper.java:294)
03-14 14:44:29.873  6763  6876 E AndroidRuntime: 	at android.os.HandlerThread.run(HandlerThread.java:67)
03-14 14:54:00.084 10203 10254 E AndroidRuntime: FATAL EXCEPTION: mrousavy/VisionCamera.main
03-14 14:54:00.084 10203 10254 E AndroidRuntime: Process: com.mrousavy.camera.example, PID: 10203
03-14 14:54:00.084 10203 10254 E AndroidRuntime: java.lang.IllegalStateException
03-14 14:54:00.084 10203 10254 E AndroidRuntime: 	at android.media.MediaRecorder.stop(Native Method)
03-14 14:54:00.084 10203 10254 E AndroidRuntime: 	at com.mrousavy.camera.core.RecordingSession.stop(RecordingSession.kt:109)
03-14 14:54:00.084 10203 10254 E AndroidRuntime: 	at com.mrousavy.camera.core.RecordingSession._init_$lambda$0(RecordingSession.kt:78)
03-14 14:54:00.084 10203 10254 E AndroidRuntime: 	at com.mrousavy.camera.core.RecordingSession.$r8$lambda$enBPf4dA0r-MTevnd6ChW75ZCDg(Unknown Source:0)
03-14 14:54:00.084 10203 10254 E AndroidRuntime: 	at com.mrousavy.camera.core.RecordingSession$$ExternalSyntheticLambda0.onError(Unknown Source:2)
03-14 14:54:00.084 10203 10254 E AndroidRuntime: 	at android.media.MediaRecorder$EventHandler.handleMessage(MediaRecorder.java:1615)
03-14 14:54:00.084 10203 10254 E AndroidRuntime: 	at android.os.Handler.dispatchMessage(Handler.java:106)
03-14 14:54:00.084 10203 10254 E AndroidRuntime: 	at android.os.Looper.loopOnce(Looper.java:205)
03-14 14:54:00.084 10203 10254 E AndroidRuntime: 	at android.os.Looper.loop(Looper.java:294)
03-14 14:54:00.084 10203 10254 E AndroidRuntime: 	at android.os.HandlerThread.run(HandlerThread.java:67)
03-14 14:54:06.457 10592 10641 E AndroidRuntime: FATAL EXCEPTION: mrousavy/VisionCamera.main
03-14 14:54:06.457 10592 10641 E AndroidRuntime: Process: com.mrousavy.camera.example, PID: 10592
03-14 14:54:06.457 10592 10641 E AndroidRuntime: java.lang.IllegalStateException
03-14 14:54:06.457 10592 10641 E AndroidRuntime: 	at android.media.MediaRecorder.stop(Native Method)
03-14 14:54:06.457 10592 10641 E AndroidRuntime: 	at com.mrousavy.camera.core.RecordingSession.stop(RecordingSession.kt:109)
03-14 14:54:06.457 10592 10641 E AndroidRuntime: 	at com.mrousavy.camera.core.RecordingSession._init_$lambda$0(RecordingSession.kt:78)
03-14 14:54:06.457 10592 10641 E AndroidRuntime: 	at com.mrousavy.camera.core.RecordingSession.$r8$lambda$enBPf4dA0r-MTevnd6ChW75ZCDg(Unknown Source:0)
03-14 14:54:06.457 10592 10641 E AndroidRuntime: 	at com.mrousavy.camera.core.RecordingSession$$ExternalSyntheticLambda0.onError(Unknown Source:2)
03-14 14:54:06.457 10592 10641 E AndroidRuntime: 	at android.media.MediaRecorder$EventHandler.handleMessage(MediaRecorder.java:1615)
03-14 14:54:06.457 10592 10641 E AndroidRuntime: 	at android.os.Handler.dispatchMessage(Handler.java:106)
03-14 14:54:06.457 10592 10641 E AndroidRuntime: 	at android.os.Looper.loopOnce(Looper.java:205)
03-14 14:54:06.457 10592 10641 E AndroidRuntime: 	at android.os.Looper.loop(Looper.java:294)
03-14 14:54:06.457 10592 10641 E AndroidRuntime: 	at android.os.HandlerThread.run(HandlerThread.java:67)
03-14 15:00:49.325 12563 12615 E AndroidRuntime: FATAL EXCEPTION: mrousavy/VisionCamera.main
03-14 15:00:49.325 12563 12615 E AndroidRuntime: Process: com.mrousavy.camera.example, PID: 12563
03-14 15:00:49.325 12563 12615 E AndroidRuntime: java.lang.IllegalStateException
03-14 15:00:49.325 12563 12615 E AndroidRuntime: 	at android.media.MediaRecorder.stop(Native Method)
03-14 15:00:49.325 12563 12615 E AndroidRuntime: 	at com.mrousavy.camera.core.RecordingSession.stop(RecordingSession.kt:109)
03-14 15:00:49.325 12563 12615 E AndroidRuntime: 	at com.mrousavy.camera.core.RecordingSession._init_$lambda$0(RecordingSession.kt:78)
03-14 15:00:49.325 12563 12615 E AndroidRuntime: 	at com.mrousavy.camera.core.RecordingSession.$r8$lambda$enBPf4dA0r-MTevnd6ChW75ZCDg(Unknown Source:0)
03-14 15:00:49.325 12563 12615 E AndroidRuntime: 	at com.mrousavy.camera.core.RecordingSession$$ExternalSyntheticLambda0.onError(Unknown Source:2)
03-14 15:00:49.325 12563 12615 E AndroidRuntime: 	at android.media.MediaRecorder$EventHandler.handleMessage(MediaRecorder.java:1615)
03-14 15:00:49.325 12563 12615 E AndroidRuntime: 	at android.os.Handler.dispatchMessage(Handler.java:106)
03-14 15:00:49.325 12563 12615 E AndroidRuntime: 	at android.os.Looper.loopOnce(Looper.java:205)
03-14 15:00:49.325 12563 12615 E AndroidRuntime: 	at android.os.Looper.loop(Looper.java:294)
03-14 15:00:49.325 12563 12615 E AndroidRuntime: 	at android.os.HandlerThread.run(HandlerThread.java:67)
03-14 15:00:58.950 12977 13023 E AndroidRuntime: FATAL EXCEPTION: mrousavy/VisionCamera.main
03-14 15:00:58.950 12977 13023 E AndroidRuntime: Process: com.mrousavy.camera.example, PID: 12977
03-14 15:00:58.950 12977 13023 E AndroidRuntime: java.lang.IllegalStateException
03-14 15:00:58.950 12977 13023 E AndroidRuntime: 	at android.media.MediaRecorder.stop(Native Method)
03-14 15:00:58.950 12977 13023 E AndroidRuntime: 	at com.mrousavy.camera.core.RecordingSession.stop(RecordingSession.kt:109)
03-14 15:00:58.950 12977 13023 E AndroidRuntime: 	at com.mrousavy.camera.core.RecordingSession._init_$lambda$0(RecordingSession.kt:78)
03-14 15:00:58.950 12977 13023 E AndroidRuntime: 	at com.mrousavy.camera.core.RecordingSession.$r8$lambda$enBPf4dA0r-MTevnd6ChW75ZCDg(Unknown Source:0)
03-14 15:00:58.950 12977 13023 E AndroidRuntime: 	at com.mrousavy.camera.core.RecordingSession$$ExternalSyntheticLambda0.onError(Unknown Source:2)
03-14 15:00:58.950 12977 13023 E AndroidRuntime: 	at android.media.MediaRecorder$EventHandler.handleMessage(MediaRecorder.java:1615)
03-14 15:00:58.950 12977 13023 E AndroidRuntime: 	at android.os.Handler.dispatchMessage(Handler.java:106)
03-14 15:00:58.950 12977 13023 E AndroidRuntime: 	at android.os.Looper.loopOnce(Looper.java:205)
03-14 15:00:58.950 12977 13023 E AndroidRuntime: 	at android.os.Looper.loop(Looper.java:294)
03-14 15:00:58.950 12977 13023 E AndroidRuntime: 	at android.os.HandlerThread.run(HandlerThread.java:67)
```

### Camera Device

```json
n\a
```

### Device

Pixel 7 - Android 14, Pixel XL4 - Android 13

### VisionCamera Version

3.9.1

### Can you reproduce this issue in the VisionCamera Example app?

Yes, I can reproduce the same issue in the Example app here

### Additional information

- [ ] I am using Expo
- [ ] I have enabled Frame Processors (react-native-worklets-core)
- [X] I have read the [Troubleshooting Guide](https://react-native-vision-camera.com/docs/guides/troubleshooting)
- [X] I agree to follow this project's [Code of Conduct](https://github.com/mrousavy/react-native-vision-camera/blob/main/CODE_OF_CONDUCT.md)
- [X] I searched for [similar issues in this repository](https://github.com/mrousavy/react-native-vision-camera/issues) and found none.

**Comment by mrousavy** [comment]
Hey! I just released a new V4 beta ([v4.0.0-beta.7](https://github.com/mrousavy/react-native-vision-camera/releases/tag/v4.0.0-beta.7)) where I fixed a bunch of issues! 💪🚀 Can you test that and see if that fixes the issue for you? 😅 

```
yarn add react-native-vision-camera@beta
```

You might need to increase the `compileSdk` to the newest Android SDK (34) if you are behind.

Or in your case, just try running the example app on the `v4` branch.

**Comment by SK-bool** [comment]
I have noticed that it doesn't crash anymore.
The code scanner works 50/50. 
Once the camera is launched the code scanner doesn't seem to be working.
But for example if you point it at a qr code it works(sometimes).

**Comment by mrousavy** [comment]
Ok, fixed that bug in latest beta!

**Comment by mrousavy** [comment]
Can you try with latest beta (10) to see if the codescanner works? @SK-bool

**Comment by SK-bool** [comment]
Yes, I just tested it, and the code scanner works! I have tested it on two phones and noticed that after the code is scanned, and I go back to the camera, the camera view is somewhat stretched. However, if I navigate to another screen from the camera and then return, the preview resets, and it is back to normal.

**Comment by SK-bool** [comment]
So: **code scanned -> navigates to a result screen -> go back to camera = stretched**

**Comment by UlianaVarlamova** [comment]
Hi,
I've tested CameraApp on [v4.0.0-beta.7](https://github.com/mrousavy/react-native-vision-camera/releases/tag/v4.0.0-beta.7) and CameraApp doesn’t crash anymore but I got errors in metro log. In the app I also can see the following errors. 
**For the first scenario:**
<img width="1318" alt="Screenshot 2024-03-28 at 15 04 25" src="https://github.com/mrousavy/react-native-vision-camera/assets/96615514/2ffe2603-96ba-4645-bc74-ddc6f983fa69">
<img width="386" alt="Screenshot 2024-03-28 at 15 18 04" src="https://github.com/mrousavy/react-native-vision-camera/assets/96615514/d0d9237b-1f83-4375-9224-f28a7907f81d">

**For the second scenario:** 
<img width="1319" alt="Screenshot 2024-03-28 at 15 04 02" src="https://github.com/mrousavy/react-native-vision-camera/assets/96615514/dcc78af7-fcb6-449f-b3f9-ca31bd852041">
<img width="488" alt="Screenshot 2024-03-28 at 15 19 04" src="https://github.com/mrousavy/react-native-vision-camera/assets/96615514/5717714f-2bf7-4255-bcb9-fce9213bc760">
<img width="388" alt="Screenshot 2024-03-28 at 15 19 25" src="https://github.com/mrousavy/react-native-vision-camera/assets/96615514/42e4527d-871d-47aa-a1c0-24e924d4803a">
<img width="388" alt="Screenshot 2024-03-28 at 15 19 46" src="https://github.com/mrousavy/react-native-vision-camera/assets/96615514/38f5d025-25d2-4cf1-a5ef-bcb73ef72729">

Are these errors expected?

**Comment by mrousavy** [comment]
No those errors are not expected - how did you test that? On `v4` branch? What device?

**Comment by UlianaVarlamova** [comment]
@mrousavy  
Source taken from: https://github.com/mrousavy/react-native-vision-camera/releases/tag/v4.0.0-beta.7
Device: Pixel 7 - Android 14,

**Comment by mrousavy** [comment]
Hey - I think this issue has been fixed in [VisionCamera 4.0.0](https://github.com/mrousavy/react-native-vision-camera/releases/tag/v4.0.0). 🥳 

Please try V4 and let me know if you still experience this issue;
- if not, please consider [💖 **sponsoring me on GitHub** 💖](https://github.com/sponsors/mrousavy/) to support the development of VisionCamera and thank me for my time spent on fixing bugs and building new features.
- if you still see this issue, please comment and we can re-open this. But please update your native logs with the native (Xcode/Android Studio) logs from running VisionCamera V4 so I can investigate this.
